nflow-tests/src/main/java/io/nflow/tests/demo/workflow/DemoWorkflow.java

```diff
@@ -41,7 +42,7 @@ public String getDescription() {
   }
 
   public DemoWorkflow() {
-    super(DEMO_WORKFLOW_TYPE, State.begin, State.error);
+    super(DEMO_WORKFLOW_TYPE, State.begin, State.error, new WorkflowSettings.Builder().setWakeupParentWhenFinished(true).build());
```

Comment:
I believe the idea of the wakup parent was to avoid changing the child workflows. This just changes the code from the step to constructor, which is clearly better. But it would be good if parent could enable the flag on the child so that any workflow could be used as bulk child.

Reply:
Remove wakeup flag from settings, added new state type wait, if parent is in wait state wake it up when child finishes